mifosng-provider/src/main/java/org/mifosplatform/portfolio/charge/serialization/ChargeDefinitionCommandFromApiJsonDeserializer.java

```diff
@@ -261,9 +272,17 @@ public void validateForUpdate(final String json) {
         }
 
         if (this.fromApiJsonHelper.parameterExists("chargeCalculationType", element)) {
+            final Integer chargeTimeType = this.fromApiJsonHelper.extractIntegerSansLocaleNamed("chargeTimeType", element);
             final Integer chargeCalculationType = this.fromApiJsonHelper.extractIntegerNamed("chargeCalculationType", element,
                     Locale.getDefault());
             baseDataValidator.reset().parameter("chargeCalculationType").value(chargeCalculationType).notNull().inMinMaxRange(1, 5);
+            if(chargeTimeType!=null&&ChargeTimeType.TRANCHE_DISBURSEMENT.getValue().equals(chargeTimeType)){
+                baseDataValidator.reset().parameter("chargeCalculationType").value(chargeCalculationType).isOneOfTheseValues(new Object[]{ChargeCalculationType.FLAT.getValue(),ChargeCalculationType.PERCENT_OF_DISBURSEMENT_AMOUNT.getValue()});;
+            }
+            if(chargeCalculationType!=null&&(!ChargeCalculationType.FLAT.getValue().equals(chargeCalculationType)&&
+                    !ChargeCalculationType.PERCENT_OF_DISBURSEMENT_AMOUNT.getValue().equals(chargeCalculationType))){
+                baseDataValidator.reset().parameter("chargeTimeType").value(chargeTimeType).isNotOneOfTheseValues(new Object[]{ChargeTimeType.TRANCHE_DISBURSEMENT.getValue()});
```

Comment:
@VenuShyam  as we had discussed the other day, since all parameters for update are optional, you can choose to  either only change the chargetimetype or the chargecalculation type. Your validation would not be complete unless also know the existing state of the charge (i.e this validation needs to be done after updating the charge)